{"pr_number": 12659, "pr_title": "Improve logging for tier mgmt task", "pr_author": "jiacheliu3", "pr_createdAt": "2020-12-17T08:34:59Z", "pr_url": "https://github.com/Alluxio/alluxio/pull/12659", "timeline": [{"oid": "32e66447c986cd41ad25419da84e69b27f5d20f3", "url": "https://github.com/Alluxio/alluxio/commit/32e66447c986cd41ad25419da84e69b27f5d20f3", "message": "add unit test for failed tier move", "committedDate": "2020-12-17T08:30:05Z", "type": "commit"}, {"id": "MDI0OlB1bGxSZXF1ZXN0UmV2aWV3Q29tbWVudDU0NDkxMTMwNQ==", "url": "https://github.com/Alluxio/alluxio/pull/12659#discussion_r544911305", "body": "This comment is from https://github.com/Alluxio/alluxio/commit/530c4c3d5781a386b962c1cc16d24b3d5b249055 it doesn't make sense here anymore. We have changed how the space allocation works largely. When the move task failed, it can be due to many more reasons than just a transient error here, for example: \r\n1) The target location is full and this move operation does not evict\r\n2) Eviction failed since the target is full of pinned blocks\r\n3) We evict on the target location but the free space is stolen by someone else\r\n4) maybe an unfound bug from concurrency\r\n\r\nBasically this TODO can be removed.", "bodyText": "This comment is from 530c4c3 it doesn't make sense here anymore. We have changed how the space allocation works largely. When the move task failed, it can be due to many more reasons than just a transient error here, for example:\n\nThe target location is full and this move operation does not evict\nEviction failed since the target is full of pinned blocks\nWe evict on the target location but the free space is stolen by someone else\nmaybe an unfound bug from concurrency\n\nBasically this TODO can be removed.", "bodyHTML": "<p dir=\"auto\">This comment is from <a class=\"commit-link\" data-hovercard-type=\"commit\" data-hovercard-url=\"https://github.com/Alluxio/alluxio/commit/530c4c3d5781a386b962c1cc16d24b3d5b249055/hovercard\" href=\"https://github.com/Alluxio/alluxio/commit/530c4c3d5781a386b962c1cc16d24b3d5b249055\"><tt>530c4c3</tt></a> it doesn't make sense here anymore. We have changed how the space allocation works largely. When the move task failed, it can be due to many more reasons than just a transient error here, for example:</p>\n<ol dir=\"auto\">\n<li>The target location is full and this move operation does not evict</li>\n<li>Eviction failed since the target is full of pinned blocks</li>\n<li>We evict on the target location but the free space is stolen by someone else</li>\n<li>maybe an unfound bug from concurrency</li>\n</ol>\n<p dir=\"auto\">Basically this TODO can be removed.</p>", "author": "jiacheliu3", "createdAt": "2020-12-17T08:48:15Z", "path": "core/server/worker/src/main/java/alluxio/worker/block/TieredBlockStore.java", "diffHunk": "@@ -362,10 +362,8 @@ public void moveBlock(long sessionId, long blockId, BlockStoreLocation oldLocati\n       }\n       return;\n     }\n-    // TODO(bin): We are probably seeing a rare transient failure, maybe define and throw some\n-    // other types of exception to indicate this case.", "originalCommit": "32e66447c986cd41ad25419da84e69b27f5d20f3", "replyToReviewId": null, "replies": null, "type": "inlineReview"}, {"oid": "34daec2ba7c47f442ce64bd68245647d804893b1", "url": "https://github.com/Alluxio/alluxio/commit/34daec2ba7c47f442ce64bd68245647d804893b1", "message": "whitespace", "committedDate": "2020-12-17T08:48:32Z", "type": "commit"}]}